apps/freeablo/fa_main.cpp

```diff
@@ -103,7 +100,7 @@ bool dataFilesSetUp(const Settings::Settings& settings)
 
     for (const auto& hash : hashes)
     {
-        mpqValid = md5Equal(hash, mpqDigest);
+        mpqValid = md5Equal(hash, digest);
```

Comment:
Looks like it only checks if the last (eu version) hash is correct..

Reply:
What do you mean? It's working here.

Reply:
Looks like it needs `if (mpqValid) break;` in that loop